test/src/unit-capi-array.cc

```diff
@@ -1511,28 +1440,22 @@ TEST_CASE_METHOD(
 TEST_CASE_METHOD(
     ArrayFx,
     "C API: Test opening array at timestamp, writes",
-    "[capi][array][open-at][writes]") {
+    "[capi][array][open-at][writes][rest-fails][sc-42722]") {
   // TODO: refactor for each supported FS.
   std::string temp_dir = fs_vec_[0]->temp_dir();
+  std::string array_name = fs_vec_[0]->is_rest() ? "tiledb://unit/" : "";
```

Comment:
I see your pain, I also don't like this but at least it's noise just for old tests. In new tests using `VfsTestSetup` you get this through `array_uri` method. Prepending it `SupportedFsS3::temp_dir` wouldn't work unfortunately afaiu because of all the cleanup/create etc operations we do directly on S3. I will look into moving the `array_uri` method in `SupportedFsS3` class to make it less ugly. Wait for my next patch please.